### ISSUE TITLE
Systemd integration

### ISSUE THREAD

**Issue by JulienBalestra** [DataDog/datadog-agent#846]
On systemd platform, it could be useful to use [sd_notify](https://www.freedesktop.org/software/systemd/man/systemd-notify.html).

This will improve the integration of the agent with Systemd and allow to create a datadog agent unit with the following directive:
```
[Service]
Type=notify
```
Then, any other application with a dependency on the Datadog agent could use with leverage the following directive without writing any custom wait-for-agent:
```
[Unit]
Requires=datadog-agent.service
After=datadog-agent.service
```

An example of this implementation is [etcd](https://github.com/coreos/etcd/blob/24c718f7dea45a0f7e2f0688c938fc3a9a34560a/etcdmain/etcd.go#L163-L168)

**Comment by masci** [comment]
Agreed this would be useful, I already have a card in our backlog to implement the readiness protocol but patches are welcome 😜

**Comment by jacjtito** [comment]
Closing this issue automatically.
